include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,402 @@
+			static constexpr auto maxChannels = 4u;
+
+			#ifdef _IRR_DEBUG
+			memset(scratchMemory, 0, state->scratchMemoryByteSize);
+			#endif // _IRR_DEBUG
+
+			const core::vector3du32_SIMD scratchByteStrides = [&]()
+			{
+				const auto trueExtent = vector3du32_SIMD(state->extent.width, state->extent.height, state->extent.depth);
+
+				switch (currentChannelCount)
+				{
+					case 1:
+					{
+						return TexelBlockInfo(asset::E_FORMAT::EF_R64_SFLOAT).convert3DTexelStridesTo1DByteStrides(trueExtent);
+					}
+
+					case 2:
+					{
+						return TexelBlockInfo(asset::E_FORMAT::EF_R64G64_SFLOAT).convert3DTexelStridesTo1DByteStrides(trueExtent);
+					}
+
+					case 3:
+					{
+						return TexelBlockInfo(asset::E_FORMAT::EF_R64G64B64_SFLOAT).convert3DTexelStridesTo1DByteStrides(trueExtent);
+					}
+					case 4:
+					{
+						return TexelBlockInfo(asset::E_FORMAT::EF_R64G64B64A64_SFLOAT).convert3DTexelStridesTo1DByteStrides(trueExtent);
+					}
+				}
+			}();
+			const auto scratchTexelByteSize = scratchByteStrides[0];
+
+			const auto&& [copyInBaseLayer, copyOutBaseLayer, copyLayerCount] = std::make_tuple(state->inBaseLayer, state->outBaseLayer, state->layerCount);
+			state->layerCount = 1u;
+
+			auto resetState = [&]()
+			{
+				state->inBaseLayer = copyInBaseLayer;
+				state->outBaseLayer = copyOutBaseLayer;
+				state->layerCount = copyLayerCount;
+			};
+
+			for (uint16_t w = 0u; w < copyLayerCount - copyInBaseLayer; ++w) 
+			{
+				std::array<decodeType, maxChannels> minDecodeValues = {};
+				std::array<decodeType, maxChannels> maxDecodeValues = {};
+
+				{
+					const auto inData = reinterpret_cast<const uint8_t*>(state->inImage->getBuffer()->getPointer());
+					const auto blockDims = asset::getBlockDimensions(state->inImage->getCreationParameters().format);
+					
+					auto decode = [&](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos) -> void
+					{
+						auto newReadBlockPos = (readBlockPos - decltype(readBlockPos)(state->inOffset.x, state->inOffset.y, state->inOffset.z)); // not sure
+
+						auto localOutPos = newReadBlockPos * blockDims;
+						auto* inDataAdress = inData + readBlockArrayOffset;
+						const size_t scratchOffset = asset::IImage::SBufferCopy::getLocalByteOffset(core::vector3du32_SIMD(localOutPos.x, localOutPos.y, localOutPos.z), scratchByteStrides);
+						decodeType* outDataAdress = (decodeType*)(reinterpret_cast<uint8_t*>(scratchMemory) + scratchOffset);
+
+						constexpr uint8_t maxPlanes = 4;
+						const void* inSourcePixels[maxPlanes] = { inDataAdress, nullptr, nullptr, nullptr };
+
+						for (auto blockY = 0u; blockY < blockDims.y; blockY++)
+							for (auto blockX = 0u; blockX < blockDims.x; blockX++)
+							{
+								decodeType decodeBuffer[maxChannels] = {};
+
+								asset::decodePixelsRuntime(inFormat, inSourcePixels, decodeBuffer, blockX, blockY);
+								memcpy(outDataAdress, decodeBuffer, scratchTexelByteSize);
```

Comment:
you cant just copy every time into same output address

Reply:
corrected in following commit